src/test/ui/associated-types/defaults-suitability.rs

```diff
@@ -0,0 +1,18 @@
+// compile-fail
+
+#![feature(associated_type_defaults)]
+
+struct NotClone;
+
+trait Tr {
+    //~^ ERROR the trait bound `NotClone: std::clone::Clone` is not satisfied
+    type Ty: Clone = NotClone;
+}
+
+trait Foo<T> {
+    type Bar: Clone = Vec<T>;
+    // FIXME this one's broken for some reason
+    // `<std::vec::Vec<T> as std::clone::Clone>` does show up in the added bounds though?
```

Comment:
Not sure why this isn't getting rejected properly. Maybe what I've done effectively gets treated as additional where-clauses on the trait:

```rust
trait Foo<T> where Vec<T>: Clone {
 ...
}
```

That would also explain the accidental interaction with the `trivial_bounds` feature below. When that feature is enabled, the above error stops getting emitted.

Maybe I did the check in the wrong place. Will investigate tomorrow.